### PR TITLE
Fix python methods, expose samples_to_pandas, integrate logging

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

**Comment by tpoterba** [comment]
Added logging support (logs no longer written to stderr)

**Comment by tpoterba** [comment]
Can you take another look, Cotton?  This PR:
 - sets up the conf as we did in Main (fixes the issues Konrad has had in the last days)
 - sets up logging properly
 - handles exceptions correctly.  All new methods that call into scala VDS operations need to wrap in try/except

File: src/main/scala/org/broadinstitute/hail/driver/package.scala [hunk 1]
```diff
@@ -39,4 +46,50 @@ package object driver {
 
     CountResult(vds.nSamples, nVariants, nCalled)
   }
+
+  def configure(sc: SparkContext, logFile: String, quiet: Boolean, append: Boolean,
```
Comment (cseed) [inline src/main/scala/org/broadinstitute/hail/driver/package.scala]: This should be called from Main.

File: python/pyhail/context.py [hunk 2]
```diff
@@ -1,7 +1,22 @@
 import pyspark
 
 from pyhail.dataset import VariantDataset
-from pyhail.java import jarray, scala_object
+from pyhail.java import jarray, scala_object, scala_package_object
+from py4j.protocol import Py4JJavaError
+
+
+class FatalError(Exception):
+    """:class:`.FatalError`
```
Comment (cseed) [inline python/pyhail/context.py]: This should be a 1-line description.

File: python/pyhail/context.py [hunk 3]
```diff
@@ -10,7 +25,10 @@ class HailContext(object):
     :param SparkContext sc: The pyspark context.
     """
 
-    def __init__(self, sc):
+    def __init__(self, sc=None, log=None, quiet=False, append=False,
```
Comment (cseed) [inline python/pyhail/context.py]: `__init__` parameters are documented in the class doc string.  Steal the text from the Main Options object.

File: python/pyhail/context.py [hunk 4]
```diff
@@ -25,24 +43,48 @@ def __init__(self, sc):
...
+            self.jsc,
+            log,
+            quiet,
+            append,
+            parquet_compression,
+            block_size,
+            branching_factor,
+            tmp_dir)
 
         logger = sc._jvm.org.apache.log4j
```
Comment (cseed) [inline python/pyhail/context.py]: You can remove this log4j stuff.  I think it is superseded by the Main configuration.

File: python/pyhail/context.py (hunk 3 above)
Comment (cseed) [inline python/pyhail/context.py]: Why not have `log = "hail.log"` like in the Main Options?  Then the log/quiet test below is unnecessary.

File: python/pyhail/context.py [hunk 5]
```diff
@@ -25,24 +43,48 @@ def __init__(self, sc):
 
         self.jsc.hadoopConfiguration().set(
```
Comment (cseed) [inline python/pyhail/context.py]: Put this in `configure()`.

File: python/pyhail/dataset.py [hunk 6]
```diff
@@ -1,12 +1,17 @@
...
 import pyspark
+from py4j.protocol import Py4JJavaError
+
 
 class VariantDataset(object):
     def __init__(self, hc, jvds):
         self.hc = hc
         self.jvds = jvds
 
+    def _raise_py4j_exception(self, e):
```
Comment (cseed) [inline python/pyhail/dataset.py]: Is this worth it?

File: src/test/scala/org/broadinstitute/hail/io/compress/BGzipCodecSuite.scala [hunk 7]
```diff
@@ -44,7 +44,7 @@ class TestFileInputFormat extends hd.mapreduce.lib.input.TextInputFormat {
 
 class BGzipCodecSuite extends SparkSuite {
   @Test def test() {
-    sc.hadoopConfiguration.set("io.compression.codecs", "org.apache.hadoop.io.compress.DefaultCodec,org.broadinstitute.hail.io.compress.BGzipCodec,org.apache.hadoop.io.compress.GzipCodec")
+    sc.hadoopConfiguration.setLong("mapreduce.input.fileinputformat.split.minsize", 1L)
```
Comment (cseed) [inline src/test/scala/org/broadinstitute/hail/io/compress/BGzipCodecSuite.scala]: I'm going to merge this is in, but why this change?

Comment (tpoterba) [inline src/test/scala/org/broadinstitute/hail/io/compress/BGzipCodecSuite.scala]: We now call `driver.configure` in the @BeforeClass method on SparkSuite.  This sets the codecs correctly, but also sets the min size to 1M.  The bgzip files generated in the test are smaller than 1M, so we need to remove the minimum to verify correct behavior
